LeanSAT/Reflect/Tactics/SatDecide.lean

```diff
@@ -63,29 +64,31 @@ def mkTemp : IO System.FilePath := do
   let out ← IO.Process.output { cmd := "mktemp" }
   return out.stdout.trim
 
+def LratCert.ofFile (lratPath : System.FilePath) : IO LratCert := do
+  let lines ← IO.FS.lines lratPath
+  -- This is just a sanity check to verify that the proof does indeed parse.
+  -- The parsing relevant for the reflection proof happens in the reflection term.
+  if LRAT.parseLRATProof lines |>.isNone then
+    throw <| IO.userError "SAT solver produced invalid LRAT"
+  -- XXX String.intercalate wit Array
```

Comment:
Is cleanup required here?

Reply:
I marked that as XXX because we go from an `Array String` to a `List String` to a `String` and later we go back to a `List` then an `Array`. But it is implemented that way because I didn't find `String.intercalate` and `String.splitOn` that return/act on `Array`. Do we have such a thing?